src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs

```diff
@@ -13,6 +13,313 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+                  // sequence point: Write(x != y ? 1U : 0);
+                  IL_0098:  ldarg.0
+                  IL_0099:  ldarg.1
+                  IL_009a:  ceq
+                  IL_009c:  ldc.i4.0
+                  IL_009d:  ceq
+                  IL_009f:  call       "void System.Console.Write(uint)"
+                  // sequence point: Write(x != y ? 1L : 0);
+                  IL_00a4:  ldarg.0
+                  IL_00a5:  ldarg.1
+                  IL_00a6:  ceq
+                  IL_00a8:  ldc.i4.0
+                  IL_00a9:  ceq
+                  IL_00ab:  conv.i8
+                  IL_00ac:  call       "void System.Console.Write(long)"
+                  // sequence point: Write(x != y ? 1UL : 0);
+                  IL_00b1:  ldarg.0
+                  IL_00b2:  ldarg.1
+                  IL_00b3:  ceq
+                  IL_00b5:  ldc.i4.0
+                  IL_00b6:  ceq
+                  IL_00b8:  conv.i8
+                  IL_00b9:  call       "void System.Console.Write(ulong)"
+                  // sequence point: Write(x != y ? (nint)1 : 0);
+                  IL_00be:  ldarg.0
+                  IL_00bf:  ldarg.1
+                  IL_00c0:  ceq
+                  IL_00c2:  ldc.i4.0
+                  IL_00c3:  ceq
+                  IL_00c5:  conv.i
+                  IL_00c6:  conv.i8
+                  IL_00c7:  call       "void System.Console.Write(long)"
+                  // sequence point: Write(x != y ? 1 : (nuint)0);
+                  IL_00cc:  ldarg.0
+                  IL_00cd:  ldarg.1
+                  IL_00ce:  ceq
+                  IL_00d0:  ldc.i4.0
+                  IL_00d1:  ceq
+                  IL_00d3:  conv.i
+                  IL_00d4:  conv.u8
+                  IL_00d5:  call       "void System.Console.Write(ulong)"
+                  // sequence point: }
+                  IL_00da:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem(61483, "https://github.com/dotnet/roslyn/issues/61483")]
+        public void Branchless_NonBinaryCondition()
+        {
+            // public static class C { public static bool M() => -1; }
+            var source1 = """
+                .class public auto ansi abstract sealed beforefieldinit C
+                    extends System.Object
+                {
+                    .method public hidebysig static bool M () cil managed
+                    {
+                        .maxstack 8
+                        ldc.i4.m1
+                        ret
+                    }
+                }
+                """;
+            var source2 = """
+                System.Console.WriteLine(D.M1());
+                System.Console.WriteLine(D.M2());
+
+                class D
+                {
+                    public static int M1() => C.M() ? 1 : 0;
+                    public static int M2() => C.M() ? 0 : 1;
+                }
+                """;
+            var comp = CreateCompilationWithIL(source2, source1);
```

Comment:
>); [](http://example.com/codeflow?start=63&length=2)

```, options: TestOptions.DebugExe``` for clarity?

Reply:
The default is `ReleaseExe`, I assume you mean that.